base_layer/wallet/src/base_node_service/service.rs

```diff
@@ -223,137 +213,116 @@ where
+            .reorg_info
+            .ok_or_else(|| BaseNodeServiceError::RpcError("Tip info has no reorg info".to_string()))?;
 
-    /// Handles the response from the connected base node.
-    async fn handle_base_node_response(
-        &mut self,
-        response: DomainMessage<proto::BaseNodeServiceResponse>,
-    ) -> Result<(), BaseNodeServiceError>
-    {
-        let DomainMessage::<_> { inner: message, .. } = response;
-
-        let (found, remaining): (Vec<RequestMetadata>, Vec<RequestMetadata>) = self
-            .requests
-            .iter()
-            .partition(|&&r| r.request_key == message.request_key);
-
-        if !found.is_empty() {
-            trace!(target: LOG_TARGET, "Handle base node response message: {:?}", message);
-
-            let now = Utc::now().naive_utc();
-            let time_sent = found.first().unwrap().sent;
-            let millis = (now - time_sent).num_milliseconds() as u64;
-            let latency = Duration::from_millis(millis);
-
-            match message.response {
-                Some(proto::base_node_service_response::Response::ChainMetadata(chain_metadata)) => {
-                    trace!(target: LOG_TARGET, "Chain Metadata response {:?}", chain_metadata);
-                    debug!(target: LOG_TARGET, "Base node latency: {}ms", millis);
-                    let metadata: ChainMetadata = chain_metadata
-                        .try_into()
-                        .map_err(BaseNodeServiceError::InvalidBaseNodeResponse)?;
-
-                    // store chain metadata in the wallet db
-                    self.db.set_chain_metadata(metadata.clone()).await?;
-
-                    let state = BaseNodeState {
-                        is_synced: Some(message.is_synced),
-                        chain_metadata: Some(metadata),
-                        updated: Some(now),
-                        latency: Some(latency),
-                        online: OnlineState::Online,
-                    };
-                    self.publish_event(BaseNodeEvent::BaseNodeState(state.clone()));
-                    self.set_state(state);
-                },
-                _ => {
-                    trace!(
+        if let Some(reorg_info_current) = self.state.reorg_info.clone() {
+            if reorg_info_current.last_reorg_best_block != reorg_info_new.last_reorg_best_block {
+                let mut transaction_service = self.transaction_service.clone();
+                let mut output_manager_service = self.output_manager_service.clone();
+                task::spawn(async move {
+                    info!(
                         target: LOG_TARGET,
-                        "Received a base node response currently unaccounted for: {:?}",
-                        message
+                        "Chain reorg detected, restarting transaction validation"
                     );
-                },
-            }
+                    if let Err(e) = transaction_service
```

Comment:
Should these not be propagated with `?`

Reply:
My opinion is that it is good enough to read in the log that there was an error here, as re-validation is important but not critical, and not to stop the rest of the code from executing, which is much more important/critical.